implementations/rust/ockam/ockam_core/src/vault/types.rs

```diff
@@ -0,0 +1,167 @@
+use cfg_if::cfg_if;
+use serde::{Deserialize, Serialize};
+use zeroize::Zeroize;
+
+/// Curve25519 private key length
+pub const CURVE25519_SECRET_LENGTH: usize = 32;
+/// Curve25519 public key length
+pub const CURVE25519_PUBLIC_LENGTH: usize = 32;
+/// P256 private key length
+pub const P256_SECRET_LENGTH: usize = 32;
+/// P256 public key length
+pub const P256_PUBLIC_LENGTH: usize = 65;
+/// AES256 private key length
+pub const AES256_SECRET_LENGTH: usize = 32;
+/// AES128 private key length
+pub const AES128_SECRET_LENGTH: usize = 16;
+
+cfg_if! {
```

Comment:
@antoinevg I think it's outdated approach, should we instead use compat?
P.S. This is out of scope of this PR